### ISSUE TITLE
angular_1_router should not depend on ES6 shims (e.g., string.startsWith)

### ISSUE THREAD

**Issue by Josh68** [angular/angular#7701]
**Note: for support questions, please use one of these channels:** https://github.com/angular/angular/blob/master/CONTRIBUTING.md#question. This repository's issues are reserved for feature requests and bug reports.

* **I'm submitting a ... **
[x ] bug report
[ ] feature request
[ ] support request => Please do not submit support request here, see note at the top of this template.

* **Do you want to request a *feature* or report a *bug*?**

Report a bug

* **What is the current behavior?**

angular_1_router throws errors in IE11 (at least) because of the inclusion of the String.prototype.startsWith method (if not anything else)

* **If the current behavior is a bug, please provide the steps to reproduce and if possible a minimal demo of the problem** via
https://plnkr.co or similar (you can use this template as a starting point: http://plnkr.co/edit/tpl:AvJOMERrnz94ekVua0u5).

build the demo locally (or create a project using the router) and run on IE11: https://github.com/brandonroberts/angularjs-component-router

* **What is the expected behavior?**

Should work in any browser that Angular 1.5.x supports

* **What is the motivation / use case for changing the behavior?**

To use the angular_1_router (ngcomponentrouter) in an Angular 1.5.x application without any ES6 shims

* **Please tell us about your environment:**

- Angular version: 2.0.0-beta.X
- Browser: [all | Chrome XX | Firefox XX | IE XX | Safari XX | Mobile Chrome XX | Android X.X Web Browser | iOS XX Safari | iOS XX UIWebView | iOS XX WKWebView ]
- Language: [all | TypeScript X.X | ES6/7 | ES5 | Dart]

- Angular 1.5.1
- IE 11
- ES5

* **Other information** (e.g. detailed explanation, stacktraces, related issues, suggestions how to fix, links for us to have context, eg. stackoverflow, gitter, etc)

Add polyfill for startsWith (or anything else ES6-specific) or, preferably, substitute ES5 syntax alternatives (e.g., routePath.substr(0,1) === '/' for routePath.startsWith('/'))

**Comment by btford** [comment]
@petebacondarwin can you take a look into this?

**Comment by petebacondarwin** [comment]
Will do

**Comment by geroyche** [comment]
i believe it's related...
this part
 `loader: () => $injector.invoke(loader)`
in [https://github.com/angular/angular/blob/master/modules/angular1_router/src/module_template.js](https://github.com/angular/angular/blob/master/modules/angular1_router/src/module_template.js)
breaks my linter, and grunt-uglify.

**Comment by mhevery** [comment]
won't fix
